src/wasm/module-decoder.cc

```diff
@@ -116,10 +116,17 @@ class ModuleDecoder : public Decoder {
                 {nullptr, 0, 0, 0, 0, 0, 0, false, false});
             WasmFunction* function = &module->functions->back();
             DecodeFunctionInModule(module, function, verify_functions);
-
-            if (ok() && verify_functions) {
-              if (!function->external)
+          }
+          if (ok() && verify_functions) {
```

Comment:
I think it's better if instead of making a second pass, any errors from validating internal functions get propagated to the error_pc and error_pt of this decoder. That will terminate the loop above sooner. We should probably also completely get rid of `result_`, since we do a `toResult()` on the end of the decode function above.